### DISCUSSION TITLE
Background Fetch and sync with cache.

### DISCUSSION THREAD

**Discussion by rajasekarm** [TanStack/query#1495] [Ideas] [answered]
Use case

Imagine Data is already fetched and cached, there are few data which is always changing.  So if we have an option to return the cached data and trigger a query in the background and sync the data in cache will be useful.

```JS
  const { data, isBackgroundSyncing} = useQuery("repoData", () => {}, {backgroundSync: true});
```
isBackgroundSyncing helps to indicate the user that update is happening in the data.

**Comment by tannerlinsley** [comment, marked as answer]
You can already do this by setting a staleTime on your query option and using the isFetching property on the resulting query info object.
